### PR TITLE
Import Callable from collections.abc

### PR DESCRIPTION
**Pull request by gabe-l-hart**
## Description

This PR attempts to address https://github.com/py4j/py4j/issues/473

### CODE REVIEW

File: py4j-python/src/py4j/clientserver.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 
 from __future__ import unicode_literals, absolute_import
 
-from collections import deque, Callable
+from collections import deque
+from collections.abc import Callable
```
Comment (HyukjinKwon) [inline py4j-python/src/py4j/clientserver.py:13]: The problem was that it breaks Python 2 so I used the deprecated one. Probably we can just remove the usage of `Callable`, and check if it is a weakref reference.

Comment (gabe-l-hart) [reply py4j-python/src/py4j/clientserver.py:13]: That makes perfect sense! Sorry I missed that. I'll to get a python 2 compatible version up tomorrow.

Comment (HyukjinKwon) [reply py4j-python/src/py4j/clientserver.py:13]: Nah, thanks for addressing this issue man.

Comment (gabe-l-hart) [reply py4j-python/src/py4j/clientserver.py:13]: According to [the internet](https://stackoverflow.com/a/624939), it looks like the builtin `callable` should be a solution that works for all 2.X and >=3.2. Is that a safe range for `py4j` at this point?

Comment (gabe-l-hart) [reply py4j-python/src/py4j/clientserver.py:13]: I've verified locally on `2.7`, `3.9`, and `3.10`. I've also verified that in `3.10` an `ImportError` is raised on `from collections import Callable`

Comment (HyukjinKwon) [reply py4j-python/src/py4j/clientserver.py]: Yeah it broke python 3.10 support. Or we can even just remove the assert for now.

**Comment by HyukjinKwon** [comment]
Actually https://github.com/py4j/py4j/pull/475 suggested the fix .. let me merge that one and close this one - will add you as a co-author of https://github.com/py4j/py4j/pull/475.

Thanks for investigation @gabe-l-hart!

**Pull request closed without merging**